### PR TITLE
Publishing documentation

### PR DESCRIPTION
**Pull request by galleon**
This PR is the sibling of #72. It does the following:
  - Push release documentation on GH pages under `version/x.y.z` where `x.y.z` is the release number
  - Push latest master documentation on GH pages (intermediate master documentations can be found as artefacts in their respective build workflows)

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -270,3 +271,14 @@ jobs:
 
       - name: generate documentation
         run: yarn global add vuepress && yarn install && yarn docs:build && touch docs/.vuepress/dist/.nojekyll
+
+      - name: Deploy documentation in root folder on GH pages 🚀
+        if: github.ref == 'refs/heads/master'
+        uses: JamesIves/github-pages-deploy-action@releases/v3
```
[the remark this answers is not in the document]
Comment (galleon) [reply .github/workflows/build.yml:277]: Bumped github-pages-deploy-action to v4.1.5 - and closing #71

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -281,8 +281,15 @@ jobs:
...
     needs: [deploy]
     runs-on: ubuntu-latest
+    env:
+      DOCS_VERSION_PATH: /
 
     steps:
+      - name: Update DOCS_VERSION_PATH
+        if: ${{ startsWith(github.ref, 'refs/tags/') }}
+        run: |
+            echo DOCS_VERSION_PATH=/version/$(echo $GITHUB_REF | cut -d / -f 3 | cut -d v -f 2-)/ >> $GITHUB_ENV
```
[the remark this answers is not in the document]
Comment (galleon) [reply .github/workflows/release.yml:291]: Should be ok

File: .github/workflows/build.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (dbarbier) [reply .github/workflows/build.yml:277]: It seems that this action has not been bumped.

File: .github/workflows/release.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (dbarbier) [reply .github/workflows/release.yml:291]: ???

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -316,13 +316,11 @@ jobs:
...
+        uses: JamesIves/github-pages-deploy-action@releases@v4.1.5
         with:
           ACCESS_TOKEN: ${{ secrets.gh_access_token }}
           BRANCH: gh-pages # The branch the action should deploy to.
           FOLDER: docs/.vuepress/dist # The folder the action should deploy.
           TARGET_FOLDER: ${{ env.DOCS_VERSION_PATH }} # The folder the action should deploy to.
-          GIT_CONFIG_EMAIL: guillaume.alleon@gmail.com
           COMMIT_MESSAGE: publish documentation
           CLEAN: true
-          SINGLE_COMMIT: true
```
Comment (dbarbier) [inline .github/workflows/release.yml:320]: You must explain in the commit message why SINGLE_COMMIT is dropped.  This is certainly not a maintenance action.

**Review by dbarbier — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (galleon) [reply .github/workflows/release.yml:291]: GITHUB_REF being of the form 'refs/tags/vx.y.z' then DOCS_VERSION_PATH=/version/x.y.z/

[the remark this answers is not in the document]
Comment (dbarbier) [reply .github/workflows/release.yml:291]: Sure it works, I thought that my proposed changes were simpler.  Anyway I do not care much.

File: .github/workflows/release.yml (hunk 3 above)
Comment (dbarbier) [reply .github/workflows/release.yml:320]: Okay, now let us talk about the CLEAN keyword.  At the moment, master and releases will delete each other, which is not very nice.
We could set CLEAN to false in both workflows, but this sounds suboptimal.  I believe that this should work:
 * set CLEAN to true in build.yml with CLEAN_EXCLUDE: ["version/*"]
 * set CLEAN to false in release.yml, there is normally nothing to clean there

**Comment by dbarbier** [comment]
Checks do not run, your workflow files have errors.

File: .github/workflows/release.yml (hunk 3 above)
Comment (galleon) [reply .github/workflows/release.yml:320]: done - `CLEAN` is `true` by default

**Comment by galleon** [comment]
> Checks do not run, your workflow files have errors.

Indeed fixed

File: .github/workflows/release.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (galleon) [reply .github/workflows/release.yml]: Oh sorry I did not understand your initial comment, I was looking for an error in my proposal.
Changed with adaptation to keep same result.

**Review by dbarbier — APPROVED** [review]
LGTM.
Be aware that as soon as this PR is merged, online documentation will be replaced by master.  If this is not intended, some migration plan has to be discussed.

**Comment by dbarbier** [comment]
Please squash your commits into 1 or 2 commits

**Pull request merged**